tools/trimmomatic.wdl

```diff
@@ -0,0 +1,151 @@
+        Int? trailing
+        Int window_size
+        Int window_quality
+        Boolean phred64
+    }
+
+    command <<<
+        ls -l /usr/local/share/trimmomatic-0.36-5/adapters/
+        trimmomatic SE -threads ~{threads} \
+        -trimlog trimlog.txt ~{input_file} output.fq.gz \
+        ~{if (phred64) then "-phred64" else "-phred33"} \
+        ILLUMINACLIP:~{illumina_clip} \
+        ~{if defined(leading) then "LEADING:~{leading}" else ""} \
+        ~{if defined(trailing) then "TRAILING:~{trailing}" else ""} \
+        SLIDINGWINDOW:~{window_size}:~{window_quality} \
+        MINLEN:~{minlen} \
+        ~{if defined(crop) then "CROP:~{crop}" else ""} \
+        ~{if defined(headcrop) then "HEADCROP:~{headcrop}" else ""} \
+        ; mv output.fq.gz ~{outfile_name}
+    >>>
+
+    output {
+        File trimmed_reads = "~{outfile_name}"
+    }
+
+    runtime {
+        memory: "16 GB"
+        disks: "100 GB"
+        container: "quay.io/biocontainers/trimmomatic:0.36--5"
+        cpu: 8
+        maxRetries: 0
+    }
+}
+
+task trim_paired_end {
+    meta {
+        description: "Performs paired end trimming of reads using the Trimmomatic tool."
+        outputs: {
+            trimmed_reads_forward: "The forward trimmed paired-end read in FASTQ format."
+            trimmed_reads_reverse: "The reverse trimmed paired-end read in FASTQ format."
+        }
+    }
+
+    parameter_meta {
+        input_file_forward: "Forward input FASTQ format file to run Trimmomatic on."
+        input_file_reverse: "Reverse input FASTQ format file to run Trimmomatic on."
+        output_name: "Output FASTQ format file of trimmed reads."
+        leading: "Cut bases off the start of a read, if below a threshold quality."
+        trailing: "Cut bases off the end of a read, if below a threshold quality."
+        window_size: "Specifies the number of bases to average across for the sliding window."
+        window_quality: "Specifies the average quality required in the sliding window."
+        minlen: "Drop the read if it is below a specified length."
+    }
+
+    input {
+        File input_file_forward
+        File input_file_reverse
+        String outfile_name_forward = (
+            basename(input_file_forward, ".fastq.gz") + ".forward.trimmed.fastq.gz"
+        )
+        String outfile_name_reverse = (
+            basename(input_file_reverse, ".fastq.gz") + ".reverse.trimmed.fastq.gz"
+        )
+         String prefix = sub(
+            basename(input_file_forward),
+            "([_\\.][rR][12])?(\\.subsampled)?\\.(fastq|fq)(\\.gz)?$",
+            ""
+        )
+        String illumina_clip
+        Int? crop
+        Int? headcrop
+        Int? leading
+        Int minlen
+        Int threads = 8
+        Int? trailing
+        Int window_size
+        Int window_quality
+        Boolean phred64
+    }
+
+    command <<<
+        ls -l /usr/local/share/trimmomatic-0.36-5/adapters/
+        trimmomatic PE -phred33 -threads ~{threads} \
```

Comment:
I think these tasks should be combined. You can dynamically select the subcommand with `~{if (defined(<read_two>)) then "PE" else "SE"}`